kombu/transport/redis.py

```diff
@@ -905,14 +906,18 @@ def _connparams(self, asynchronous=False):
             'socket_connect_timeout': self.socket_connect_timeout,
             'socket_keepalive': self.socket_keepalive,
             'socket_keepalive_options': self.socket_keepalive_options,
+            'health_check_interval': self.health_check_interval,
         }
 
         conn_class = self.connection_class
+
+        # If the connection class does not support the `health_check_interval`
+        # argument then remove it.
         if (
             hasattr(conn_class, '__init__') and
-            accepts_argument(conn_class.__init__, 'health_check_interval')
+            not accepts_argument(conn_class.__init__, 'health_check_interval')
         ):
-            connparams['health_check_interval'] = DEFAULT_HEALTH_CHECK_INTERVAL
```

Comment:
The old logic would add the DEFAULT_HEALTH_CHECK_INTERVAL if the connection class did support the argument, but now we are not defaulting it here and we are popping it from the connparams -- would popping it  like that blow up if health_check_interval were not already in the connparams?  I question why we are inverting this logic pattern.

Reply:
@matteius I inverted it because when you read this entire method the pattern seems to be to remove items not add them conditionally. Not that I necessarily think its a good programming pattern but I thought it should at least be consistent with how all other arguments are work here.